src/Action/CancelAction.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Answear\Payum\PayU\Action;
+
+use Answear\Payum\PayU\Enum\OrderStatus;
+use Answear\Payum\PayU\Model\Model;
+use Answear\Payum\PayU\Request\OrderRequestService;
+use Answear\Payum\PayU\Util\PaymentHelper;
+use Payum\Core\Action\ActionInterface;
+use Payum\Core\Exception\RequestNotSupportedException;
+use Payum\Core\Model\PaymentInterface;
+use Payum\Core\Request\Cancel;
+use Psr\Log\LoggerInterface;
+use Webmozart\Assert\Assert;
+
+class CancelAction implements ActionInterface
+{
+    private const FINAL_STATUSES = [
+        OrderStatus::Completed,
+        OrderStatus::Canceled,
+    ];
+
+    public function __construct(
+        private OrderRequestService $orderRequestService,
+        private LoggerInterface $logger
+    ) {
+    }
+
+    /**
+     * @param Cancel $request
+     */
+    public function execute($request): void
+    {
+        RequestNotSupportedException::assertSupports($this, $request);
+
+        $model = Model::ensureArrayObject($request->getModel());
+
+        $payment = PaymentHelper::ensurePayment($request->getFirstModel());
+        Assert::notNull($payment, 'Payment must be set on cancel action.');
+        $orderId = PaymentHelper::getOrderId($model, $payment);
+        Assert::notEmpty($orderId, 'OrderId must be set on cancel action.');
+
+        try {
+            if (!$this->canCancelPayment($model, $payment)) {
+                return;
+            }
+
+            $this->orderRequestService->cancel($model->orderId(), PaymentHelper::getConfigKey($model, $payment));
+        } catch (\Throwable $throwable) {
+            $this->logger->critical('Cannot cancel payment.', ['exception' => $throwable]);
```

Comment:
You can't catch whole exceptions and only log it, we need to know it something happened. Throw custom exception `CancelPaymentException` and pass parent to it

Reply:
I removed the `try catch` completely, it is not needed here